sound/soc/sof/pcm.c

```diff
@@ -318,6 +318,16 @@ static int sof_pcm_trigger(struct snd_pcm_substream *substream, int cmd)
 	snd_sof_pcm_platform_trigger(sdev, substream, cmd);
 
 	/* send IPC to the DSP */
+	/*
+	 * The FE can be triggered in a non-atomic environment because
+	 * if the non-atomic property of pcm is true, ALSA core will
+	 * set the trigger action be in non-atomic environment.
+	 * The non-atomic property is set in the DAI link when parsing
+	 * topology and copied into the PCM fields.
+	 * The BE doesn't set the non-atomic property because BE's no_pcm
+	 * property is set to true. This means ALSA core will not use
+	 * BE's non-atomic property.
```

Comment:
Sorry @libinyang, you are describing the actions when the .non-atomic field is set in the dailink but not the rationale behind setting this .non-atomic field.
Again we need to explain WHY the FE needs this .non-atomic field set and the BE does not.

Reply:
How about explaining "WHY FE PCM is set .non-atomic" in topology.c, and just mention here that PCMs are .non-atomic so sending IPC here is safe?